packages/expo-in-app-purchases/src/InAppPurchases.ts

```diff
@@ -0,0 +1,121 @@
+export { default as ExpoInAppPurchasesView } from './ExpoInAppPurchasesView';
+
+const validTypes = {
+  INAPP: 'inapp',
+  SUBS: 'subs',
+}
+const billingResponseCodes = ExpoInAppPurchases.responseCodes;
+const purchaseStates = ExpoInAppPurchases.purchaseStates;
+
+export const constants = {
+  billingResponseCodes,
+  purchaseStates,
+  validTypes,
+}
+
+let connected = false;
+
+export async function connectToAppStoreAsync(): Promise<QueryResponse> {
+  console.log('calling connectToAppStoreAsync from TS');
+  if (connected) {
+    throw new ConnectionError('Already connected to App Store');
+  }
+
+  connected = true;
+  return await ExpoInAppPurchases.connectToAppStoreAsync();
+}
+
+export async function queryPurchasableItemsAsync(itemList: string[]): Promise<QueryResponse> {
+  console.log('calling queryPurchasableItemsAsync from TS');
+  if (!connected) {
+    throw new ConnectionError('Must be connected to App Store');
+  }
+  if (Platform.OS === 'android') {
+    // On Android you have to pass in the item type so we will combine the results of both inapp and subs
+    const { responseCode, results } = await ExpoInAppPurchases.queryPurchasableItemsAsync(validTypes.INAPP, itemList);
+    if (responseCode == billingResponseCodes.OK) {
+      const subs = await ExpoInAppPurchases.queryPurchasableItemsAsync(validTypes.SUBS, itemList);
+      subs.results.forEach(result => {
+        results.push(result);
+      });
+    }
+    return { responseCode, results };
+  }
+
+  return await ExpoInAppPurchases.queryPurchasableItemsAsync(itemList);
+}
+
+export async function queryPurchaseHistoryAsync(refresh?: boolean): Promise<QueryResponse> {
+  console.log('calling queryPurchaseHistoryAsync from TS');
+  if (!connected) {
+    throw new ConnectionError('Must be connected to App Store');
+  }
+  if (refresh && Platform.OS === 'android') {
+    const { responseCode, results } = await ExpoInAppPurchases.queryPurchaseHistoryAsync(validTypes.INAPP);
+    if (responseCode === billingResponseCodes.OK) {
+      const subs = await await ExpoInAppPurchases.queryPurchaseHistoryAsync(validTypes.SUBS);
+      subs.results.forEach(result => {
+        results.push(result);
+      });
+    }
+    return { responseCode, results };
+  }
+  return await ExpoInAppPurchases.queryPurchaseHistoryAsync(null);
+}
+
+export async function purchaseItemAsync(itemId: string, oldItem?: string): Promise<QueryResponse> {
+  console.log('calling purchaseItemAsync from TS');
+  if (!connected) {
+    throw new ConnectionError('Must be connected to App Store');
+  }
+
+  // Replacing old item is only supported on Android
+  const args = Platform.OS === 'android' ? [itemId, oldItem] : [itemId];
+  return await ExpoInAppPurchases.purchaseItemAsync(...args);
+}
+
+export async function acknowledgePurchaseAsync(purchaseToken: string, consumeItem: boolean): Promise<void> {
+  // No-op if not on Android since this is not applicable
+  if (Platform.OS !== 'android') return;
```

Comment:
It might be better to check it this way:
```suggestion
  if (!Expo.acknowledgePurchaseAsync) {
    throw new UnavailabilityError('InAppPurchases', 'acknowledgePurchaseAsync');
  }
```
Also, could you please do such checks in all other methods? 🙂

Reply:
I'm all for adding that check in other methods but I think in this case specifically it might be better to no-op on iOS rather than throw an error on the off chance that a user accidentally calls it.

For example, the way I currently envision the API being used is as follows:
```javascript
const { responseCode, results } = await purchaseItemAsync(productId);
if (responseCode === constants.billingResponseCodes.OK) {
  for (const purchase of results) {
    if (!purchase.acknowledged) {
      await acknowledgePurchaseAsync(purchase.purchaseToken, true);
    }
  }
}
```
on iOS I have the `acknowledged` field always return true since it's not applicable and I expect the developer to only call `acknowledgePurchaseAsync` if a purchase is not acknowledged. However, on the off chance that they forgot to do that it should just return immediately rather than error.

I mainly just want to reduce the need to check for `Platform.OS === 'android'` for platform specific code so that the developer has less to worry about. Let me know what you think.